terraform-workshop/9100_basics/modul_basics_9100.md

```diff
@@ -253,6 +253,89 @@ Defines terraform configuration.
 
 ---
 
+## meta-parameters
+Can be applied to all resource definitions regardless of type
+
+* count (not applicable to modules)
+* depends_on
+* provider
+* lifecycle
+  * create_before_destroy
+  * prevent_destroy
+  * ignore_changes
+
+
+## count (not applicable to modules)
+```hcl
+resource "azurerm_virtual_machine" "main" {
+  count = 3 # creates 3 virtual machines
+  name = "VM-${count.index}"
+}
+```
+
+Creates multiple instances of the resource
+
+## depends_on
+```hcl
+resource "azurerm_virtual_machine" "main" {
+  # forces creation of the network interface before the VM
+  depends_on = [ "${azurerm_network_interface.main}" ]
```

Comment:
Using a dependency to a module might be more realistic as an example.

Reply:
I kinda like how this demonstrates forced recreation of the nic whereas recreating some modules might wreak more havoc. WDYT